xclim/indices/_agro.py

```diff
@@ -1524,3 +1526,118 @@ def hardiness_zones(
 
     zones = zones.assign_attrs(units="")
     return zones
+
+
+def _accumulate_intermediate(prev_E, prev_xi, curr_xs, curr_ak1):
+    """Accumulate the intermediate product based on the previous concentration and the current temperature."""
+    curr_S = np.where(prev_E < 1, prev_E, prev_E - prev_E * prev_xi)
+    return curr_xs - (curr_xs - curr_S) * np.exp(-curr_ak1)
+
+
+def _chill_portion_one_season(tas_K):
+    """Computes the chill portion for a single season based on the dynamic model on a numpy array."""
+    # Constants as described in Luedeling et al. (2009)
+    E0 = 4153.5
+    E1 = 12888.8
+    A0 = 139500
+    A1 = 2.567e18
+    SLP = 1.6
+    TETMLT = 277
+    AA = A0 / A1
+    EE = E1 - E0
+
+    ftmprt = SLP * TETMLT * (tas_K - TETMLT) / tas_K
+    sr = np.exp(ftmprt)
+    xi = sr / (1 + sr)
+    xs = AA * np.exp(EE / tas_K)
+    ak1 = A1 * np.exp(-E1 / tas_K)
+
+    inter_E = np.zeros_like(tas_K)
+    for i in range(1, tas_K.shape[-1]):
+        inter_E[..., i] = _accumulate_intermediate(
+            inter_E[..., i - 1], xi[..., i - 1], xs[..., i], ak1[..., i]
+        )
+    delta = np.where(inter_E >= 1, inter_E * xi, 0)
+
+    return delta.cumsum(axis=-1)
+
+
+def _apply_chill_portion_one_season(tas_K):
+    """Apply the chill portion function on to an xarray DataArray."""
+    tas_K = tas_K.chunk(time=-1)
+    return xarray.apply_ufunc(
+        _chill_portion_one_season,
+        tas_K,
+        input_core_dims=[["time"]],
```

Comment:
```suggestion
    return xarray.apply_ufunc(
        _chill_portion_one_season,
        tas_K,
        select_time(tas_K.time.notnull(), **indexer),
        input_core_dims=[["time"]],
```
Non working example, I'll leave the implementation to you ;).

My idea here is to pass the mask of "days to consider" to the function. By taking `tas_K.time.notnull()` I am getting a 1D array of True values along `time`. Then `select_time` puts NaN outside the subset, leaving me with True values only on the timesteps I want to consider. I'll call this `mask` in the function.